docs: Add docs for new "search" query parameter (LLC-294) (#81)
LearningLocker/docs: src/http-connection.md

You must additionally supply your Basic Auth details with each request in the `Authorization` header. Your Basic Auth details can be found under **Settings** > **Clients**. The API also accepts the following *optional* URL parameters for filtering the models returned.

- [sort](#sort-parameter) (required -  we recommend sorting by `_id` if nothing else)

> - [search](#search-parameter) (available only for User connection and for GET requests of User REST API)

Reviewer: ```suggestion
- [search](#search-parameter) (available only for user connection in Enterprise)
```

#### Sorting With Improved Performance
You may find that changing the sort parameter can vary the time it takes a query to run, especially when you have a large number of models. You can take advantage of database indexes to improve performance, more information is available about [using indexes via Mongo's documentation](https://docs.mongodb.com/manual/indexes/). If utilising indexes doesn't have the required performance improvement, you can instead utilise [BI tools](../guides-retrieving).

> ### Search Parameter

Reviewer: James will want an example request here so where it currently says "So if the value for this parameter is `exampleSearchString`, it will be transformed into filter as shown below." I think we should perhaps reword that to say "So if the value for this parameter is `exampleSearchString`, as shown in the request below, it will be transformed into filter as shown below."

> The search parameter is a simple string and **available only for User connection**. This parameter is gonna search for the match over `name` and `email` fields of User schema. So if the value for this parameter is `exampleSearchString`, it will be transformed into filter as shown below.

Reviewer: ```suggestion
The search parameter is a simple string and **available only for User connection**. This parameter will search for the match over the `name` and `email` fields of User schema. So if the value for this parameter is `exampleSearchString`, it will be transformed into filter as shown below.
```
